lib/rucio/core/replica.py

```diff
@@ -3086,18 +3371,33 @@ def get_cleaned_updated_collection_replicas(total_workers, worker_number, limit=
             schema = BASE.metadata.schema + '.'
         session.execute(text('DELETE FROM {schema}updated_col_rep A WHERE A.rowid > ANY (SELECT B.rowid FROM {schema}updated_col_rep B WHERE A.scope = B.scope AND A.name=B.name AND A.did_type=B.did_type AND (A.rse_id=B.rse_id OR (A.rse_id IS NULL and B.rse_id IS NULL)))'.format(schema=schema)))  # NOQA: E501
```

Comment:
Should this query be updated to use the data model while we're here?

Reply:
Changed this to:

```
    if session.bind.dialect.name == 'oracle':
        A = aliased(models.UpdatedCollectionReplica)
        B = aliased(models.UpdatedCollectionReplica)

        subquery = select(
            literal_column('ROWID')
        ).select_from(
            B
        ).where(
            and_(A.scope == B.scope,
                 A.name == B.name,
                 A.did_type == B.did_type,
                 or_(A.rse_id == B.rse_id,
                     and_(A.rse_id.is_(None),
                          B.rse_id.is_(None))))
        )

        stmt_del = delete(
            A
        ).where(
            literal_column('ROWID') > any_(subquery.scalar_subquery())
        ).execution_options(
            synchronize_session=False
        )
        session.execute(stmt_del)
```

Which compiles to:
```
DELETE FROM dev.updated_col_rep AS updated_col_rep_1 WHERE ROWID > ANY (SELECT ROWID
FROM dev.updated_col_rep AS updated_col_rep_2
WHERE updated_col_rep_1.scope = updated_col_rep_2.scope AND updated_col_rep_1.name = updated_col_rep_2.name AND updated_col_rep_1.did_type = updated_col_rep_2.did_type AND (updated_col_rep_1.rse_id = updated_col_rep_2.rse_id OR updated_col_rep_1.rse_id IS NULL AND updated_col_rep_2.rse_id IS NULL))
```

Reply:
cc: @mlassnig (Sorry if this is the wrong ping I think the comp leads page is out of date...)